apps/meteor/ee/app/livechat-enterprise/server/hooks/manageDepartmentUnit.ts

```diff
@@ -0,0 +1,51 @@
+import type { ILivechatDepartment, IOmnichannelBusinessUnit } from '@rocket.chat/core-typings';
+import { LivechatDepartment, LivechatUnit } from '@rocket.chat/models';
+
+import { hasAnyRoleAsync } from '../../../../../app/authorization/server/functions/hasRole';
+import { callbacks } from '../../../../../lib/callbacks';
+import { getUnitsFromUser } from '../methods/getUnitsFromUserRoles';
+
+callbacks.add(
+	'livechat.manageDepartmentUnit',
+	async ({ userId, departmentId, unitId }) => {
```

Comment:
Could we create unit tests for this? You don't need to simulate the callback executing in order to test that, it's acceptable to extract the callback function into a function and export it, so it helps you to test that.

Reply:
Done! That made me notice we were decreasing the departments count in units even when an invalid unit id was provided by an admin/manager. That's fixed :partying_face: (no action is performed when an invalid unit id is provided now)